base_layer/wallet/src/text_message_service.rs

```diff
@@ -436,3 +608,69 @@ impl TextMessageServiceApi {
         res
     }
 }
+
+mod test {
+    use crate::text_message_service::{Contact, TextMessageError, TextMessageService};
+    use tari_comms::types::{CommsPublicKey, CommsSecretKey};
+    use tari_crypto::keys::{PublicKey, SecretKey};
+
+    #[test]
+    fn test_contacts_crud() {
+        let mut rng = rand::OsRng::new().unwrap();
+
+        let secret_key = CommsSecretKey::random(&mut rng);
+        let public_key = CommsPublicKey::from_secret_key(&secret_key);
+
+        let mut tms = TextMessageService::new(public_key);
+
+        let mut contacts = Vec::new();
+
+        let screen_names = vec![
+            "Alice".to_string(),
+            "Bob".to_string(),
+            "Carol".to_string(),
+            "Dave".to_string(),
+            "Eric".to_string(),
+        ];
+        for i in 0..5 {
+            let contact_secret_key = CommsSecretKey::random(&mut rng);
+            let contact_public_key = CommsPublicKey::from_secret_key(&contact_secret_key);
```

Comment:
same nit. can replace ln 636 and ln 637 with let (contact_secret_key, contact_public_key) = CommsPublicKey::random_keypair(&mut rng);

Reply:
Done